openstack_plugin/utils.py

```diff
@@ -437,10 +437,19 @@ def get_property_by_name(property_name):
         extra_resource_config = resource_config.pop('kwargs')
         resource_config.update(extra_resource_config)
 
+    # If name property is provided for the resource but it does not contain
+    # any value, then we should generate a name based on the type of the
+    # resource
+    if 'name' in resource_config and not resource_config.get('name'):
+        name = "{0}_{1}_{2}".format(class_decl.resource_type,
+                                    ctx_node.deployment.id,
+                                    ctx_node.instance.id)
+        resource_config['name'] = name
```

Comment:
Name's should probably use `-` and not `_` since they become host names. We also need to check that out in regards to ctx instance id.

Reply:
I agree with u on this, but we the issue is with `ctx_node.instance.id` which contains `_`